### PR TITLE
Use bootstrap-timeout for controller service retry;

### PR DESCRIPTION
**Pull request by ycliuhw**
*Use bootstrap-timeout for controller service retry;*

## Checklist

 - [ ] ~Requires a [pylibjuju](https://github.com/juju/python-libjuju) change~
 - [ ] ~Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR~
 - [ ] ~Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed~
 - [x] Comments answer the question of why design decisions were made

## QA steps

```console
$ juju bootstrap gke k1 --config="bootstrap-timeout=1"
Creating Juju controller "k1" on gke/asia-southeast1
Fetching Juju Dashboard 0.3.0
Creating k8s resources for controller "controller-k1"
ERROR failed to bootstrap model: creating controller stack: creating service for controller: waiting for controller service address fully provisioned timeout
```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1920835

### CODE REVIEW

**Comment by ycliuhw** [comment]
[Default timeout for service creation in k8s bootstrap on openstack too short and not configurable Edit](https://trello.com/c/CROQ8XX0/1472-default-timeout-for-service-creation-in-k8s-bootstrap-on-openstack-too-short-and-not-configurable-edit)

File: caas/kubernetes/provider/bootstrap.go [hunk 1]
```diff
@@ -492,11 +494,10 @@ func (c *controllerStack) createControllerService() error {
 		// we do Not want bootstrap-state cmd wait instead.
 		return nil
 	}
-
 	retryCallArgs := retry.CallArgs{
-		Attempts:    60,
+		Attempts:    -1,
 		Delay:       3 * time.Second,
-		MaxDuration: 3 * time.Minute,
+		MaxDuration: c.timeout,
```
Comment (wallyworld) [inline caas/kubernetes/provider/bootstrap.go:500]: this doesn't take account of the time already elapsed till this moment
but it's better than what we have

**Review by wallyworld — APPROVED** [review]

File: caas/kubernetes/provider/bootstrap.go (hunk 1 above)
Comment (ycliuhw) [reply caas/kubernetes/provider/bootstrap.go:500]: ah, we should probably pass the abort chan from `envcontext.ProviderCallContext`, and rely on the top-level timeout strategy.

File: caas/kubernetes/provider/bootstrap.go [hunk 2]
```diff
@@ -327,9 +330,9 @@ func (c *controllerStack) Deploy() (err error) {
 			return false
 		}
 	}
-	if isDone() {
-		return bootstrap.Cancelled()
-	}
+	// if isDone() {
```
Comment (hpidcock) [inline caas/kubernetes/provider/bootstrap.go:333]: Should this just be removed?

**Review by hpidcock — APPROVED** [review]

File: caas/kubernetes/provider/bootstrap.go (hunk 2 above)
Comment (ycliuhw) [reply caas/kubernetes/provider/bootstrap.go]: Uncommented.
This broke the test because of context.Context was not mocked.

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**